packages/runtime-core/src/component.ts

```diff
@@ -894,7 +894,7 @@ export function createSetupContext(
         return shallowReadonly(instance.slots)
       },
       get emit() {
-        return (event: string, ...args: any[]) => instance.emit(event, ...args)
+        return instance.emit
```

Comment:
This is intentional so the returned emit always calls potentially overwritten instance emit.

Users can potentially cache destructured `emit` before the overwrite.